getResource_test.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestFoldersTableSlice(t *testing.T) {
+	var foldersResBody *foldersResp
+	json.Unmarshal(fakeFolderByte, &foldersResBody)
+	slice1 := []string{"707", "The Foo Folder"}
+	slice2 := []string{"708", "The Baz Folder"}
+	expectedTable := [][]string{slice1, slice2}
+	matrixTestHelper(t, foldersResBody, expectedTable)
+}
+
+func TestSitesTableSlice(t *testing.T) {
+	var sitesResBody *sitesResp
+	json.Unmarshal(fakeSitesByte, &sitesResBody)
+	slice1 := []string{"1337", "Dyer"}
+	slice2 := []string{"42", "Situation"}
+	expectedTable := [][]string{slice1, slice2}
+	matrixTestHelper(t, sitesResBody, expectedTable)
+}
+
+func TestBrowsersTableSlice(t *testing.T) {
+	var browsersRespBody *browsersResp
+	err := json.Unmarshal(fakeClientsByte, &browsersRespBody)
+	if err != nil {
+		panic(err)
+	}
+	slice1 := []string{"firefox", "Mozilla Firefox"}
+	slice2 := []string{"ie11", "Microsoft Internet Explorer 11"}
+	expectedTable := [][]string{slice1, slice2}
+	matrixTestHelper(t, browsersRespBody, expectedTable)
+}
+
+func matrixTestHelper(t *testing.T, resBody returnTable, expectedTable [][]string) {
+	actualTable := resBody.TableSlice()
+	if expectedLen, actualLen := len(expectedTable), len(actualTable); expectedLen != actualLen {
+		t.Errorf("Wrong number of matrix rows. Expected %d, got %d", expectedLen, actualLen)
+	}
+
+	for i, actualrow := range actualTable {
+		for j, actualColumn := range actualrow {
+			if expectedTable[i][j] != actualColumn {
+				t.Errorf("Unexpected matrix entry. Expected %s, got %s", expectedTable[i][j], actualColumn)
+			}
+		}
+	}
+}
+
+var fakeFolderByte = []byte(`[
```

Comment:
In general, don't name variables with the type (it's redundant and distracts from the purpose). Something like `fakeFolderResp` would probably make more sense.